opencti-platform/opencti-front/src/private/components/settings/ThemesEditor.tsx

```diff
@@ -0,0 +1,488 @@
+  const theme = useTheme<Theme>();
+  const { t_i18n } = useFormatter();
+  const [open, setOpen] = useState<boolean>(false);
+  const { about } = useLazyLoadQuery<ThemesEditorAboutQuery>(aboutQuery, {});
+  const [commit] = useApiMutation<ThemeCreatorCreateMutation>(
+    createThemeMutation,
+    undefined,
+    {
+      successMessage: `${t_i18n('Theme')} ${t_i18n('successfully created')}`,
+      errorMessage: t_i18n('Failed to import theme'),
+    },
+  );
+
+  const themeOptions = themes?.edges
+    ?.filter((node) => !!node)
+    .map((node) => ({ ...node.node })) ?? [];
+
+  const themeValidator = Yup.object().shape({
+    name: Yup.string()
+      .trim()
+      .min(2)
+      .required(t_i18n('This field is required')),
+    theme_background: Yup.string()
+      .matches(/^#[0-9a-fA-F]{6}$/)
+      .required(t_i18n('This field is required')),
+    theme_paper: Yup.string()
+      .matches(/^#[0-9a-fA-F]{6}$/)
+      .required(t_i18n('This field is required')),
+    theme_nav: Yup.string()
+      .matches(/^#[0-9a-fA-F]{6}$/)
+      .required(t_i18n('This field is required')),
+    theme_primary: Yup.string()
+      .matches(/^#[0-9a-fA-F]{6}$/)
+      .required(t_i18n('This field is required')),
+    theme_secondary: Yup.string()
+      .matches(/^#[0-9a-fA-F]{6}$/)
+      .required(t_i18n('This field is required')),
+    theme_accent: Yup.string()
+      .matches(/^#[0-9a-fA-F]{6}$/)
+      .required(t_i18n('This field is required')),
+    theme_logo: Yup.string().nullable(),
+    theme_logo_collapsed: Yup.string().nullable(),
+    theme_logo_login: Yup.string().nullable(),
+  });
+  const handleSubmitField = (id: string, name: string, value: string) => {
+    themeValidator.validateAt(name, { [name]: value }).then(() => {
+      commitMutation({
+        ...defaultCommitMutation,
+        mutation: editThemeMutation,
+        variables: { id, input: { key: name, value: value ?? '' } },
+      });
+    });
+  };
+  const handleImport = (event: FormEvent) => {
+    const inputElement = event.target as HTMLInputElement;
+    const fileReader = new FileReader();
+    const file = inputElement.files?.[0];
+    if (!file) return;
+    fileReader.readAsText(file, 'UTF-8');
+    fileReader.onload = (e) => {
+      try {
+        let targetString;
+        const target = e.target?.result;
+        if (!target) throw Error(t_i18n('No file target found'));
+        if (typeof target === 'string') {
+          targetString = target;
+        } else {
+          targetString = new TextDecoder().decode(target);
+        }
+
+        const parsedFile = JSON.parse(targetString);
+        if (!parsedFile) {
+          // JSON.parse should throw a syntax error
+          throw Error(t_i18n('Failed to parse file'));
+        }
+
+        if (parsedFile.openCTI_version !== about?.version) {
+          throw Error(t_i18n('Incompatible version'));
+        }
+        if (parsedFile.type !== 'theme') {
+          throw Error(t_i18n('Incompatible type'));
```

Comment:
Same remark, maybe add the type required as it is done for Dashboard import ?
![image](https://github.com/user-attachments/assets/ccd79c61-d308-4159-867a-2d39f1a69d3f)